iceoryx_utils/include/iceoryx_utils/cxx/forward_list.hpp

```diff
@@ -0,0 +1,406 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CXX_FORWARD_LIST_HPP_INCLUDED
+#define CXX_FORWARD_LIST_HPP_INCLUDED
+
+#include <cstdint>
+#include <iostream>
+
+#include "iceoryx_utils/platform/platform_correction.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+/// @brief  C++11 compatible uni-directional forward list implementation.
+///         Adjustments in the API were done to not use exceptions and serve the requirement of
+///         a data structure movable over shared memory.
+///         attempt to add elements to a full list will be ignored.
+///         Capacity must at least be 1, (unintended) negative initialization is rejected with compile assertion
+///         limitation: concurrency concerns have to be handled by client side.
+/// @param T type user data to be managed within list
+/// @param Capacity number of maximum list elements a client can push to the list. minimum value is '1'
+template <typename T, uint64_t Capacity>
+class forward_list
```

Comment:
What about a `using list = cxx::forward_list`?

Reply:
your idea: give the cxx::forward_list an interface which can replace any existing list implementations as smooth as possible.
my thoughts: honourable direction! One thing separating usage of list vs. forward_list are calls to 'before_begin' and 'erase_after'. One leaves more freedom of (possibly) adding a cxx::list implementation in future by not occupying a type name 'list' in forward_list already. 
Might we leave such an alias declaration to be done in user code?